jax/experimental/jax2tf/call_tf.py

```diff
@@ -0,0 +1,232 @@
+    args_tf_sig_flat = [
+        tf.TensorSpec(np.shape(a_jax), _to_tf_dtype(_dtype(a_jax)))
+        for a_jax in args_jax_flat
+    ]
+    args_tf_sig = tf.nest.map_structure(
+        lambda a_jax: tf.TensorSpec(
+            np.shape(a_jax), _to_tf_dtype(_dtype(a_jax))), args_jax)
+    func_tf_concrete = tf.function(func_tf).get_concrete_function(*args_tf_sig)
+    res_tf_sig_flat, res_treedef = tree_util.tree_flatten(
+        func_tf_concrete.structured_outputs)
+
+    res_jax_flat = call_tf_p.bind(
+        *args_jax_flat,
+        func_tf=func_tf,
+        args_treedef=args_jax_treedef,
+        args_tf_sig_flat=args_tf_sig_flat,
+        res_treedef=res_treedef,
+        res_tf_sig_flat=res_tf_sig_flat)
+    # TODO(necula): check the expected result signature
+    assert len(res_jax_flat) == len(res_tf_sig_flat)
+    return res_treedef.unflatten(res_jax_flat)
+
+  # Define the fwd and bwd custom_vjp functions
+  def make_call_vjp_fwd(*args_jax):
+    # Return the primal argument as the residual
+    return make_call(*args_jax), args_jax
+
+  def make_call_vjp_bwd(residual, ct_res):
+    args_jax = residual  # residual is the primal argument
+
+    def tf_vjp_fun(arg, ct_res):
+      """Invoke TF gradient."""
+      with tf.GradientTape(persistent=True) as tape:
+        tape.watch(arg)
+        res = func_tf(*arg)
+
+      tf.nest.assert_same_structure(res, ct_res)
+      # If the result is not a scalar, we must accumulate arguments cotangents.
+      accumulator = None  # Same structure as "arg"
+
+      def acc_ct(res_, ct_res_):
+        dres_darg = tape.gradient(
+            res_,
+            sources=arg,
+            unconnected_gradients=tf.UnconnectedGradients.ZERO)
+        tf.nest.assert_same_structure(dres_darg, arg)
+        scaled_dres_darg = tf.nest.map_structure(lambda d: d * ct_res_,
+                                                 dres_darg)
+        nonlocal accumulator
+        accumulator = (
+            scaled_dres_darg if accumulator is None
+            else tf.nest.map_structure(
+                lambda x, y: x + y, accumulator, scaled_dres_darg))
+
+      tf.nest.map_structure(acc_ct, res, ct_res)
+      return accumulator
+    # Use call_tf to call the VJP function
+    return call_tf(tf_vjp_fun)(args_jax, ct_res)
+
+  make_call.defvjp(make_call_vjp_fwd, make_call_vjp_bwd)
+  return util.wraps(func_tf)(make_call)
+
+
+call_tf_p = core.Primitive("call_tf")
+call_tf_p.multiple_results = True
+
+
+# The impl will be used in op-by-op mode and calls func_tf in TF eager mode.
+def _call_tf_impl(*args, args_treedef, func_tf, **_):
+  # TODO(necula): does this copy to the host?
+  args_tf_flat = [tf.constant(np.asarray(a)) for a in args]
+  res_tf = func_tf(*args_treedef.unflatten(args_tf_flat))
+  # TODO(necula): check the result for tree and aval
+  # TODO(necula): does this copy to the host?
+  return [jnp.asarray(np.asarray(r)) for r in tree_util.tree_flatten(res_tf)[0]]
```

Comment:
nit: `tree_util.tree_flatten(res_tf)[0]` could be `tree_util.tree_leaves(res_tf)`